buildtest/modules/util.py

```diff
@@ -1,48 +1,156 @@
-import subprocess
+from buildtest.tools.system import BuildTestCommand
 
 
 def get_all_collections():
-    """Return all module collection """
+    """Get all Lmod user collections that is retrieved by running ``module -t savelist``.
+
+     :return: Return all module collections
+     :rtype: list
+     """
+
     collections = "module -t savelist"
-    ret = subprocess.Popen(
-        collections, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE
-    )
-    out = ret.communicate()[1]
-    tmp = out.decode("utf-8").split()
+    cmd = BuildTestCommand()
+    cmd.execute(collections)
+    # The output is captured in STDERR stream.
+    out = cmd.get_error().split()
```

Comment:
I very much like the design of this command object - that we can execute, get_error, I'd even add another function to make it easy to print (preview) what command is going to be issued - this would be helpful to hook up to a debug mode.

Reply:
let me try to add a method to ``BuildTestCommand()`` to get that information.